libs/community/langchain_community/utilities/cassandra_database.py

```diff
@@ -0,0 +1,705 @@
+"""Apache Cassandra database wrapper."""
+from __future__ import annotations
+
+import re
+import traceback
+from typing import Any, Dict, List, Optional, Sequence, Tuple, Union
+
+import cassio.config
```

Comment:
It seems the dependency to cassio could be made optional when users provide the session by their own means.

Reply:
Also the cassio should be imported dynamically as it's not required in the poetry deps.